src/entity.js

```diff
@@ -542,6 +527,25 @@ export class Entity {
   }
 
   /**
+   * Validates the entity
+   *
+   * @param {string|null} propertyName Optional. The name of the property to validate. If unspecified,
+   * all properties will be validated.
+   * @param {Rule<any, any>[]|null} rules Optional. If unspecified, the rules will be looked up using
+   * the metadata for the object created by ValidationRules....on(class/object)
+   * @return {Promise<ValidationError[]>}
+   */
+  validate(propertyName, rules) {
+    // entities without validation are to be considered valid
+    if (!this.hasValidation()) {
+      return Promise.resolve([]);
+    }
+
+    return propertyName ? this.getValidator().validateProperty(this, propertyName, rules)
+                        : this.getValidator().validateObject(this, rules);
```

Comment:
the StandardValidator takes care of that. Question would be custom validator. the abstract Validator class has though
   * @param rules Optional. If unspecified, the implementation should lookup the rules for the 
hence they should also take care of that themself